repository/repository.go

```diff
@@ -106,13 +125,14 @@ type FreeTierAAT struct {
 }
 
 type GatewaySettings struct {
-	SecretKey            string              `json:"secretKey"`
-	SecretKeyRequired    bool                `json:"secreyKeyRequired"`
-	WhitelistOrigins     []string            `json:"whitelistOrigins,omitempty"`
-	WhitelistUserAgents  []string            `json:"whitelistUserAgents,omitempty"`
-	WhitelistContracts   []WhitelistContract `json:"whitelistContracts,omitempty"`
-	WhitelistMethods     []WhitelistMethod   `json:"whitelistMethods,omitempty"`
-	WhitelistBlockchains []string            `json:"whitelistBlockchains,omitempty"`
+	SecretKey           string   `json:"secretKey"`
+	SecretKeyRequired   bool     `json:"secreyKeyRequired"`
+	WhitelistOrigins    []string `json:"whitelistOrigins,omitempty"`
+	WhitelistUserAgents []string `json:"whitelistUserAgents,omitempty"`
+	// TODO: change next two whitelist to use their structs
+	WhitelistContracts   []string `json:"whitelistContracts,omitempty"`
```

Comment:
Could you take another look at this field please? The original struct is an array of `BlockchainID` as well as a list of strings called `Contracts`. The suggested replacement only maintains a list of strings.

Reply:
Yes I have a TODO comment to change it for the actual structure, the thing is that right now is saved as an array.